# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/rene/samba/linux-prio_tree_h.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./client/smbmount.c.orig	2004-11-29 19:01:01.927182208 +0100
+++ ./client/smbmount.c	2004-11-29 19:01:36.032997336 +0100
@@ -24,6 +24,11 @@
 
 #include <mntent.h>
 #include <asm/types.h>
+/* for linux-2.6.9, from linux/types.h */
+#  include <asm/types.h>
+#  ifndef pgoff_t
+#  define pgoff_t unsigned long
+#  endif
 #include <linux/smb_fs.h>
 
 extern BOOL in_client;
--- ./client/smbumount.c.orig	2004-11-29 19:02:59.018381640 +0100
+++ ./client/smbumount.c	2004-11-29 19:03:24.708476152 +0100
@@ -11,6 +11,11 @@
 
 #include <asm/types.h>
 #include <asm/posix_types.h>
+/* for linux-2.6.9, from linux/types.h */
+#  include <asm/types.h>
+#  ifndef pgoff_t
+#  define pgoff_t unsigned long
+#  endif
 #include <linux/smb.h>
 #include <linux/smb_mount.h>
 #include <linux/smb_fs.h>
